compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/ClassCodegen.kt

```diff
@@ -384,8 +353,26 @@ open class ClassCodegen protected constructor(
             return
         }
 
-        val signature = FunctionCodegen(method, this).generate().asmMethod
-        jvmSignatureClashDetector.trackMethod(method, RawSignature(signature.name, signature.descriptor, MemberKind.METHOD))
+        val node = FunctionCodegen(method, this).generate()
+        val mv = with(node) { visitor.newMethod(method.OtherOrigin, access, name, desc, signature, exceptions.toTypedArray()) }
```

Comment:
NB: cached bytecode in general case would be totally different from writen one: cause optimizations are not perfromed (visitor.newMethod adds a bunch of optimization wrappers)

Reply:
Oh, sorry it's not cached